File: problems/树/404.左叶子之和.cpp
#include <iostream>
using namespace std;
struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

int sumOfLeftLeaves(TreeNode* root) {
    if (!root) return 0;
    int total = 0;
    if (root->left && !root->left->left && !root->left->right) total += root->left->val;
    else total += sumOfLeftLeaves(root->left);
    total += sumOfLeftLeaves(root->right);
    return total;
}
int main() {
    TreeNode* root = new TreeNode(3);
    root->left  = new TreeNode(9);
    root->right = new TreeNode(20);
    root->left->left  = new TreeNode(1);
    root->left->right = new TreeNode(3);
    root->right->left  = new TreeNode(15);
    root->right->right = new TreeNode(7);
    cout << sumOfLeftLeaves(root) << endl;       // 24
    cout << sumOfLeftLeaves(new TreeNode(1)) << endl;  // 0
    cout << sumOfLeftLeaves(nullptr) << endl;           // 0
    return 0;
}

File: problems/树/404.左叶子之和.py
#!/usr/bin/env python3
"""
404. 左叶子之和
"""
from typing import Optional
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def sum_of_left_leaves(root: Optional[TreeNode]) -> int:
    if not root: return 0
    total = 0
    if root.left and not root.left.left and not root.left.right:
        total += root.left.val
    else:
        total += sum_of_left_leaves(root.left)
    total += sum_of_left_leaves(root.right)
    return total

if __name__ == "__main__":
    root1 = TreeNode(3, TreeNode(9, TreeNode(1), TreeNode(3)), TreeNode(20, TreeNode(15), TreeNode(7)))
    print(sum_of_left_leaves(root1))  # 24 (1+15+8)
    root2 = TreeNode(1)
    print(sum_of_left_leaves(root2))  # 0
    print(sum_of_left_leaves(None))  # 0
